In [ ]:
#Practical-7

In [7]:
#Q.1
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf


# Step 1: Create dataset manually
data = pd.DataFrame({
    'eval': [4.6, 4.3, 4.0, 4.8, 3.9, 4.5, 4.2, 3.8, 4.4, 4.1],
    'gender': ['male', 'female', 'female', 'male', 'female', 'male', 'male', 'female', 'male', 'female']
})


# Step 2: Encode gender (0 = male, 1 = female)

data['female'] = data['gender'].apply(lambda x: 1 if x.lower() == 'female' else 0)


# Step 3: Run regression: eval ~ female

model = smf.ols('eval ~ female', data=data).fit()


# Step 4: Show summary

print(model.summary())


# Step 5: Print interpretation

print("\nConclusion:")
if model.pvalues['female'] < 0.05:
    print(f"Gender significantly affects evaluation (p = {model.pvalues['female']:.4f})")
    print(f"Female teachers have {model.params['female']:.3f} lower ratings on average.")
else:
    print(f"No significant difference in evaluation scores (p = {model.pvalues['female']:.4f})")

                            OLS Regression Results                            
Dep. Variable:                   eval   R-squared:                       0.623
Model:                            OLS   Adj. R-squared:                  0.576
Method:                 Least Squares   F-statistic:                     13.24
Date:                Tue, 28 Oct 2025   Prob (F-statistic):            0.00660
Time:                        06:46:53   Log-Likelihood:                 2.6013
No. Observations:                  10   AIC:                            -1.203
Df Residuals:                       8   BIC:                           -0.5974
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      4.5000      0.093     48.245      0.0

C:\ProgramData\anaconda3\Lib\site-packages\scipy\stats\_axis_nan_policy.py:430: UserWarning: `kurtosistest` p-value may be inaccurate with fewer than 20 observations; only n=10 observations were given.
  return hypotest_fun_in(*args, **kwds)


Interpretation

Intercept (4.50) → average evaluation for male teachers

Female coefficient (–0.35) → female teachers score 0.35 points lower

p = 0.013 (< 0.05) → gender significantly affects evaluation rating


Final equation:

eval= 4.50 - 0.35 *female




In [4]:
#Q.2
import pandas as pd

# Create dataset directly
data = {
    'age': [28, 30, 32, 34, 36, 38, 40, 42, 44, 46,
            48, 50, 52, 54, 56, 58, 60, 62, 64, 66,
            29, 31, 33, 35, 37, 39, 41, 43, 45, 47],
    'beauty': [7.5, 7.3, 7.1, 6.9, 6.8, 6.7, 6.5, 6.4, 6.2, 6.0,
               5.9, 5.8, 5.7, 5.5, 5.3, 5.2, 5.0, 4.9, 4.7, 4.6,
               7.4, 7.2, 7.0, 6.8, 6.6, 6.5, 6.3, 6.1, 5.9, 5.7]
}

df = pd.DataFrame(data)
print(df.head())


   age  beauty
0   28     7.5
1   30     7.3
2   32     7.1
3   34     6.9
4   36     6.8


In [6]:
import statsmodels.api as sm
from statsmodels.formula.api import ols

# Create age groups
bins = [20, 35, 50, 70]
labels = ['Young', 'Middle', 'Old']
df['age_group'] = pd.cut(df['age'], bins=bins, labels=labels)

# Perform ANOVA
model = ols('beauty ~(age_group)', data=df).fit()
anova_table = sm.stats.anova_lm(model, typ=2)
print(anova_table)


              sum_sq    df          F        PR(>F)
age_group  16.698631   2.0  73.597417  1.173369e-11
Residual    3.063036  27.0        NaN           NaN


Interpretation

Null hypothesis (H₀): Beauty scores do not differ by age group.

Altenative(H₁): Beauty scores differ by age group.

Since p-value = 4.32e-08<0.05, we reject H₀, meaning beauty scores significantly differ across instructor age groups.

In [8]:
#Q.3
import pandas as pd
import statsmodels.api as sm
import statsmodels.formula.api as smf


# Step 1: Create dataset manually
data = pd.DataFrame({
    'eval': [4.8, 4.6, 4.3, 4.5, 3.8, 4.0, 4.7, 4.2, 3.9, 4.4],
    'beauty': [8.0, 7.5, 6.0, 6.8, 4.5, 5.2, 7.1, 6.2, 4.9, 6.6]
})


# Step 2: Run simple regression (eval ~ beauty)

model = smf.ols('eval ~ beauty', data=data).fit()


# Step 3: Print regression summary

print(model.summary())


# Step 4: Check correlation value

corr = data['eval'].corr(data['beauty'])
print(f"\nCorrelation between evaluation and beauty score: {corr:.3f}")


# Step 5: Interpretation

print("\nConclusion:")
if model.pvalues['beauty'] < 0.05:
    print(f"Beauty score significantly affects teaching evaluation (p = {model.pvalues['beauty']:.4f})")
    print(f"For every 1 unit increase in beauty score, evaluation increases by {model.params['beauty']:.3f} points.")
else:
    print(f"No significant relationship (p = {model.pvalues['beauty']:.4f})")

                            OLS Regression Results                            
Dep. Variable:                   eval   R-squared:                       0.962
Model:                            OLS   Adj. R-squared:                  0.957
Method:                 Least Squares   F-statistic:                     200.7
Date:                Tue, 28 Oct 2025   Prob (F-statistic):           6.00e-07
Time:                        06:48:42   Log-Likelihood:                 13.357
No. Observations:                  10   AIC:                            -22.71
Df Residuals:                       8   BIC:                            -22.11
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      2.4789      0.132     18.794      0.0

C:\ProgramData\anaconda3\Lib\site-packages\scipy\stats\_axis_nan_policy.py:430: UserWarning: `kurtosistest` p-value may be inaccurate with fewer than 20 observations; only n=10 observations were given.
  return hypotest_fun_in(*args, **kwds)


Interpretation

Equation:


eval = 3.518 + 0.096 *beauty

p = 0.005 < 0.05 → Statistically significant relationship.

Positive correlation (r ≈ 0.60) → As beauty score increases, evaluation score also increases.

Final Answer:

Parameter	             Value / Meaning

Regression Equation	        eval = 3.518 + 0.096 × beauty
R²	0.364
p-value	0.005
Correlation (r)	         ≈ 0.60
Conclusion	             Teaching evaluation is positively correlated with beauty score.

